keras/backend/numpy/image.py

```diff
@@ -13,6 +13,22 @@
 )
 
 
+def psnr(image1, image2, max_val):
+    if image1.shape != image2.shape:
+        raise ValueError(
+            "Image shapes must match for PSNR calculation. "
+            f"Received shapes: image1.shape={image1.shape}, "
+            "image2.shape={image2.shape}. "
+        )
+
+    max_val = np.float32(max_val)
+    image1 = image1.astype(np.float32)
```

Comment:
Is the float32 casting done for the log operation? In general it's preferred for ops not to cast inputs to a different dtype. What we should do is resolve the output dtype  based on `image1.dtype` and `image2.dtype`, and cast to that via `convert_to_tensor(image1, dtype=dtype)` (resp image2). Please check out how other ops with 2 inputs do this.

Reply:
But i need to use float32 but image1.dtype gives float64